src/adapter/graphql/tests/tests/test_gql_dataset_adapter_collection.rs

```diff
@@ -0,0 +1,954 @@
+                "datasetId": did,
+            }))),
+        )
+        .await;
+
+    assert!(res.is_ok(), "{res:#?}");
+    pretty_assertions::assert_eq!(
+        res.data.into_json().unwrap()["datasets"]["byId"]["asCollection"]["latest"]["entries"]
+            ["nodes"]
+            .clone(),
+        json!([{
+            "path": "/foo",
+            "ref": foo.dataset_handle.id,
+            "extraData": {},
+            "asDataset": {
+                "alias": "kamu/foo",
+            },
+        }]),
+    );
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Harness
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[oop::extend(BaseGQLDatasetHarness, base_gql_harness)]
+struct GraphQLDatasetsHarness {
+    base_gql_harness: BaseGQLDatasetHarness,
+    catalog_authorized: dill::Catalog,
+}
+
+#[bon]
+impl GraphQLDatasetsHarness {
+    #[builder]
+    pub async fn new(
+        tenancy_config: TenancyConfig,
+        mock_dataset_action_authorizer: Option<MockDatasetActionAuthorizer>,
+    ) -> Self {
+        let base_gql_harness = BaseGQLDatasetHarness::builder()
+            .tenancy_config(tenancy_config)
+            .maybe_mock_dataset_action_authorizer(mock_dataset_action_authorizer)
+            .build();
+
+        let cache_dir = base_gql_harness.temp_dir().join("cache");
+        std::fs::create_dir(&cache_dir).unwrap();
+
+        let base_catalog = dill::CatalogBuilder::new_chained(base_gql_harness.catalog())
+            .add::<RenameDatasetUseCaseImpl>()
+            .add::<DeleteDatasetUseCaseImpl>()
+            .add_value(kamu::EngineConfigDatafusionEmbeddedBatchQuery::default())
+            .add::<kamu::QueryServiceImpl>()
+            .add::<kamu::ObjectStoreRegistryImpl>()
+            .add::<kamu::ObjectStoreBuilderLocalFs>()
+            .add_value(kamu::EngineConfigDatafusionEmbeddedIngest::default())
+            .add::<kamu::EngineProvisionerNull>()
+            .add::<kamu::DataFormatRegistryImpl>()
+            .add::<kamu::PushIngestPlannerImpl>()
+            .add::<kamu::PushIngestExecutorImpl>()
+            .add::<kamu::PushIngestDataUseCaseImpl>()
+            .add::<kamu_adapter_http::platform::UploadServiceLocal>()
+            .add_value(kamu::domain::ServerUrlConfig::new_test(None))
+            .add_value(kamu::domain::FileUploadLimitConfig::new_in_bytes(100))
+            .add_value(kamu::domain::CacheDir::new(cache_dir))
+            .build();
+
+        let (_catalog_anonymous, catalog_authorized) = authentication_catalogs(&base_catalog).await;
+
+        // TODO: Yuck
+        let catalog_authorized = catalog_authorized
+            .builder_chained()
+            .add_value(kamu_adapter_http::AccessToken {
+                token: "<secret>".into(),
+            })
+            .build();
```

Comment:
Q: do we need a token here since `Tracing` has been added to `schema_quiet`?

Reply:
I left a comment to remove this token and the ugly dependency from GQL to HTTP adapter once we move direct upload/download test into e2e layer.

Reply:
Never mind - I went ahead and moved direct uploads to e2e tests